lib/location/location.c

```diff
@@ -161,9 +209,51 @@ void location_config_defaults_set(
 	}
 
 	memset(config, 0, sizeof(struct location_config));
-	config->methods_count = methods_count;
-	config->timeout = 300 * MSEC_PER_SEC; /* 5 minutes */
+	config->interval = CONFIG_LOCATION_REQUEST_DEFAULT_INTERVAL;
+	config->timeout = CONFIG_LOCATION_REQUEST_DEFAULT_TIMEOUT;
 	config->mode = LOCATION_REQ_MODE_FALLBACK;
+
+	/* Handle Kconfig's for method priorities */
+	if (method_types == NULL) {
+		methods_count = 0;
+		if (IS_ENABLED(CONFIG_LOCATION_REQUEST_DEFAULT_METHOD_FIRST_GNSS)) {
+			method_types_tmp[0] = LOCATION_METHOD_GNSS;
+			methods_count++;
+		} else if (IS_ENABLED(CONFIG_LOCATION_REQUEST_DEFAULT_METHOD_FIRST_WIFI)) {
+			method_types_tmp[0] = LOCATION_METHOD_WIFI;
+			methods_count++;
+		} else if (IS_ENABLED(CONFIG_LOCATION_REQUEST_DEFAULT_METHOD_FIRST_CELLULAR)) {
+			method_types_tmp[0] = LOCATION_METHOD_CELLULAR;
+			methods_count++;
+		} else {
+			__ASSERT_NO_MSG(false);
```

Comment:
Is this assert actually necessary? Is it possible that none of the choice values is selected?

Reply:
No, maybe I should have cellular branch as the "else" branch and do an assert there.

Reply:
What do you want to assert there? If it's not GNSS or Wi-Fi, it has to be cellular. That assert could never fail.

Reply:
Not until another method is added. I left it out.